### DISCUSSION TITLE
Using vite-plugin-ssr with vite build javascript api.

### DISCUSSION THREAD

**Discussion by joaocosteira** [vikejs/vike#990] [Discussion] [answered]
Hello everyone, I'm seeking some help because I'm having some issues combining this plugin with the vite's build api function.
For simplicity's sake, I'm testing on the scaffold application (npm init vite-plugin-ssr, vue+ts option) and I simply added the prerender option to the vite config.
When I'm running vite build, everything works fine, I'm able to generate the pre rendered pages and see them by running vite preview.

But I simply created a simple script that calls the build method from the [vite's js api ](https://vitejs.dev/guide/api-javascript.html#build):
```ts
(async () => {
    await build({
        configFile: join(process.cwd(), 'vite.config.js')
    });
})()
```
Any pre rendered page is generated. Can anyone help me out figuring out what the issue?
My end result is by running the aforementioned script/function, obtaining the same result as simply running vite build command.

**Comment by brillout** [comment, marked as answer]
https://vite-plugin-ssr.com/disableAutoFullBuild is automatically set to `true` when using Vite's `build()` API.


Set it to `false` and use `vite-plugin-ssr@0.4.132-commit-36814e9` (FYI https://github.com/brillout/vite-plugin-ssr/commit/36814e995d4a7075b363af239e329200fa6373dd).

I didn't test it but it should work. Let me know if it doesn't.

**Reply by joaocosteira** [reply]
Thank you so much.
With your help I was able to programmatically pre-render everything like I wanted.

**Reply by brillout** [reply]
👍 Btw. if you're able to convince your employer: https://github.com/sponsors/brillout &mdash; it makes a big difference.

**Reply by joaocosteira** [reply]
Hello, I have to sincerely apologize but I messed up badly when I told you that everything was working correctly.
Long story short I wrote a builder.ts and a prerender.ts file, with a respective run builder and run prerender script on the package json for testing purposes.  
 Mistakenly  I ran ```build and prerenderer``` instead of  ```builder and prerenderer```, so only today when I tried to integrate these modification I saw the problem. Again I'm extremely sorry.

Here's the project ( scaffold project + vue-ts option + vite-plugin-ssr@0.4.132-commit-36814e9).

```ts
// vite.config.ts
const config: UserConfig = {
  plugins: [
    vue(), 
    ssr({
        prerender: {disableAutoRun: true},
        disableAutoFullBuild: false
    })
  ],
}
```

```ts
//prerender.ts
import path from 'path';
import { prerender } from 'vite-plugin-ssr/prerender';

( async () => {
    await prerender({
        viteConfig: {
            configFile: path.join(process.cwd(), 'vite.config.ts'),
        }
    });
})()
```

```ts
//builder.ts
import { build } from 'vite';
import path from 'path';

(async () => {
    await build({
        configFile: path.join(process.cwd(), 'vite.config.ts'),
    })
})();
```

And now if I try to run builder.ts I get the following error message:

```
computing gzip size (0)...Error: [vite-plugin-ssr@0.4.132-commit-36814e9][Bug] You stumbled upon a bug in vite-plugin-ssr's source code. Go to https://github.com/brillout/vite-plugin-ssr/issues/new and copy-paste this error; a maintainer will fix the bug (usually under 24 hours).
    at getViteBuildCliConfig ([redacted]/vite-ssr-final/node_modules/.pnpm/vite-plugin-ssr@0.4.132-commit-36814e9_vite@4.0.3/node_modules/vite-plugin-ssr/dist/cjs/node/plugin/shared/isViteCliCall.js:60:24)
    at triggerFullBuild ([redacted]/vite-ssr-final/node_modules/.pnpm/vite-plugin-ssr@0.4.132-commit-36814e9_vite@4.0.3/node_modules/vite-plugin-ssr/dist/cjs/node/plugin/plugins/autoFullBuild.js:65:69)
    at Object.handler ([redacted]/vite-ssr-final/node_modules/.pnpm/vite-plugin-ssr@0.4.132-commit-36814e9_vite@4.0.3/node_modules/vite-plugin-ssr/dist/cjs/node/plugin/plugins/autoFullBuild.js:30:31)
    at file://[redacted]/vite-ssr-final/node_modules/.pnpm/rollup@3.26.0/node_modules/rollup/dist/es/shared/node-entry.js:25312:40
    at async Promise.all (index 0)
    at async PluginDriver.hookParallel (file://[redacted]/vite-ssr-final/node_modules/.pnpm/rollup@3.26.0/node_modules/rollup/dist/es/shared/node-entry.js:25240:9)
    at async file://[redacted]/vite-ssr-final/node_modules/.pnpm/rollup@3.26.0/node_modules/rollup/dist/es/shared/node-entry.js:26552:13
    at async catchUnfinishedHookActions (file://[redacted]/vite-ssr-final/node_modules/.pnpm/rollup@3.26.0/node_modules/rollup/dist/es/shared/node-entry.js:25678:16)
    at async doBuild (file://[redacted]/vite-ssr-final/node_modules/.pnpm/vite@4.0.3_@types+node@18.11.9/node_modules/vite/dist/node/chunks/dep-0bae2027.js:44492:22)
    at async build (file://[redacted]/vite-ssr-final/node_modules/.pnpm/vite@4.0.3_@types+node@18.11.9/node_modules/vite/dist/node/chunks/dep-0bae2027.js:44321:16)
    at async file://[redacted]/vite-ssr-final/builder.ts:5:5
```

Again, I have to sincerely apology for wrongly telling that everything was working correctly, it is 100% my fault.

**Reply by brillout** [reply]
No worries and I see what's going on. I'll fix it.

**Reply by brillout** [reply]
[Fix](https://github.com/brillout/vite-plugin-ssr/commit/36ffa2b3533864a2df6abbd3f415ae6426eef593) released in `0.4.132-commit-36ffa2b`. I didn't test it but let me know if it doesn't work and I'll fix it promptly.

**Reply by joaocosteira** [reply]
Thank you so much, now I'm 100% sure sure that I've tested everything correctly and everything is up and running.
